File: AppSettingsManagerAPI/AppSettingsManagerApi.Model/Requests/CreatePermissionRequest.cs
namespace AppSettingsManagerApi.Model.Requests;

public class CreatePermissionRequest
{
    public string UserId { get; set; }
    public string SettingGroupId { get; set; }
    public PermissionLevel StartingPermissionLevel { get; set; }
    private bool NeedsApproval = true;

    public bool GetNeedsApproval()
    {
        return NeedsApproval;
    }

    public void SetNeedsApproval(bool needsApproval)
    {
        NeedsApproval = needsApproval;
    }
}

File: AppSettingsManagerUI/src/app/services/bff/models/CreatePermissionRequest.ts
import { PermissionLevel } from '../models/PermissionLevel';

export class CreatePermissionRequest {
  UserId: string;
  SettingGroupId: string;
  StartingPermissionLevel: PermissionLevel;

  constructor(userId: string, settingGroupId: string, permissionLevel: PermissionLevel){
    this.UserId = userId;
    this.SettingGroupId = settingGroupId;
    this.StartingPermissionLevel = permissionLevel;
  }
}
